### PR TITLE
hack/multi-node: self hosted etcd option

### PR DESCRIPTION
**Pull request by hongchaodeng**
Separated from #259.

We have tested it.

### CODE REVIEW

File: hack/multi-node/bootkube-up [hunk 1]
```diff
@@ -10,7 +10,7 @@ fi
 SELF_HOST_ETCD=${SELF_HOST_ETCD:-false}
 if [ ${SELF_HOST_ETCD} = "true" ]; then
     etcd_render_flags="--etcd-servers=http://10.3.0.15:2379 --storage-backend=etcd3 --experimental-self-hosted-etcd"
-    etcd_start_flags="--etcd-server=http://172.17.4.101:12379,http://10.3.0.15:2379 --experimental-self-hosted-etcd"
+    etcd_start_flags="--etcd-server=http://172.17.4.101:12379 --experimental-self-hosted-etcd"
```
Comment (aaronlevy) [inline hack/multi-node/bootkube-up]: But this won't work if the etcd pod ever leaves this one node...

Comment (aaronlevy) [inline hack/multi-node/bootkube-up]: And in fact, this is the temporary etcd pod -- how does this work at all?

Comment (hongchaodeng) [inline hack/multi-node/bootkube-up]: See my explanation in https://github.com/kubernetes-incubator/bootkube/pull/259#issuecomment-272307260

Comment (xiang90) [inline hack/multi-node/bootkube-up]: basically, the temp api server only contacts the boot member. once the boot member dies, bootkube dies. we know the boot member dies by contacting the first in-cluster etcd member directly. (alternative, we know the boot member dies if temp api server cannot contact to the boot member anymore)

there is no point that the temp api server needs to contact the first in cluster member i think.

**Pull request merged**